contrib/factory-config-vpn

```diff
@@ -0,0 +1,69 @@
+#!/bin/sh -e
+
+## This is an OnChanged handler that can handle wireguard VPN config
+## files created by fioctl.
+
+delete_vpn() {
+	if nmcli connection | grep -q ^factory-vpn ; then
+		nmcli connection delete factory-vpn
+	fi
+}
+
+for x in server client ; do
+	if [ ! -f /var/run/secrets/wireguard-$x ] ; then
+		echo "No config found for this $x"
+		delete_vpn
+		exit 0
+	fi
+done
+
+. /var/run/secrets/wireguard-server
+if [ "$enabled" = "0" ] ; then
+	echo "VPN access disabled for this factory"
+	delete_vpn
+	exit 0
+fi
+server_pub=${pubkey}
+unset enabled
+
+. /var/run/secrets/wireguard-client
+if [ "$enabled" = "0" ] ; then
+	echo "VPN access disabled for this device"
+	delete_vpn
+	exit 0
+fi
+
+[ -d /etc/wireguard ] || mkdir /etc/wireguard
+
+delete_vpn
+
+nmfile="/etc/NetworkManager/system-connections/factory-vpn.nmconnection"
+cat > $nmfile <<EOF
+[connection]
+id=factory-vpn0
```

Comment:
I'm just not completely sure if having the file name differently than id would cause issues when calling nmcli directly, but I'm assuming you tested with id with 0 here.

Reply:
It wasn't working well. I just pushed a change to make the vpnid a variable. This should give the flexibility for someone to (ab)use this.

Reply:
Great, this makes it more flexible as well, +1.